LoopUp.Siesta/SiestaClient.cs

```diff
@@ -9,23 +9,89 @@ namespace LoopUp.Siesta
+        /// </summary>
+        /// <param name="httpClient">A pre-configured <see cref="HttpClient"/>.</param>
+        /// <param name="siestaClientConfigurationOptions">Configuration options for a <see cref="SiestaClient"/>.</param>
+        protected SiestaClient(HttpClient httpClient, SiestaClientConfigurationOptions siestaClientConfigurationOptions)
+        {
+            this.client = httpClient;
+            this.correlationIdHeaderName = siestaClientConfigurationOptions.RequestHeaderCorrelationIdKey;
         }
 
         /// <inheritdoc />
         public async Task<Task> SendAsync(SiestaRequest siestaRequest)
         {
-            var response = await this.client.SendAsync(siestaRequest.GenerateRequestMessage());
+            return await this.SendRequestWithNoExpectedContent(siestaRequest);
+        }
+
+        /// <inheritdoc />
+        public async Task<Task> SendAsync(SiestaRequest siestaRequest, string currentCorrelationId)
+        {
+            return await this.SendRequestWithNoExpectedContent(siestaRequest, currentCorrelationId);
+        }
+
+        /// <inheritdoc />
+        public async Task<TReturn> SendAsync<TReturn>(SiestaRequest<TReturn> siestaRequest)
+        {
+            return await this.SendRequestWithExpectedContent<TReturn>(siestaRequest.GenerateRequestMessage());
+        }
+
+        /// <inheritdoc />
+        public async Task<TReturn> SendAsync<TReturn>(SiestaRequest<TReturn> siestaRequest, string currentCorrelationId)
+        {
+            return await this.SendRequestWithExpectedContent<TReturn>(siestaRequest.GenerateRequestMessage(), currentCorrelationId);
+        }
+
+        /// <inheritdoc />
+        public async Task<TReturn> SendAsync<TReturn>(SiestaPatchRequest<TReturn> siestaPatchRequest)
+        {
+            var originalResource = await this.SendRequestWithExpectedContent<TReturn>(siestaPatchRequest.GenerateGetRequestMessage());
+
+            return await this.SendRequestWithExpectedContent<TReturn>(
+                siestaPatchRequest.GeneratePatchRequestMessage(originalResource));
+        }
+
+        /// <inheritdoc />
+        public async Task<TReturn> SendAsync<TReturn>(SiestaPatchRequest<TReturn> siestaPatchRequest, string currentCorrelationId)
+        {
+            var originalResource = await this.SendRequestWithExpectedContent<TReturn>(siestaPatchRequest.GenerateGetRequestMessage(), currentCorrelationId);
+
+            return await this.SendRequestWithExpectedContent<TReturn>(
+                siestaPatchRequest.GeneratePatchRequestMessage(originalResource), currentCorrelationId);
+        }
+
+        private async Task<Task> SendRequestWithNoExpectedContent(SiestaRequest siestaRequest, string? currentCorrelationId = null)
+        {
+            var requestMessage = siestaRequest.GenerateRequestMessage();
+
+            if (currentCorrelationId is not null)
+            {
+                if (this.correlationIdHeaderName is null)
```

Comment:
💦  Is there a standardised header for correlation? or even something that ASP.Net uses that works out of the box? If so I'd prefer it to use that by default - if there isn't then I think it's fine to throw.

Reply:
I am not sure, I will take a look

Reply:
I think though I would prefer not to default to something. I think if you have not specified the header, but you are then trying to use one, it shows that it has either been configured incorrectly, or there is a lack of knowledge of how to use it. Either way I think it should throw in this case